mycore-solr/src/main/java/org/mycore/solr/search/MCRConditionTransformer.java

```diff
@@ -231,21 +227,22 @@ private static StringBuilder stripPlus(StringBuilder sb) {
     }
 
     public static SolrQuery getSolrQuery(@SuppressWarnings("rawtypes") MCRCondition condition, List<MCRSortBy> sortBy,
-        int maxResults) {
+        int maxResults, String returnFields) {
         String queryString = getQueryString(condition);
         SolrQuery q = applySortOptions(new SolrQuery(queryString), sortBy);
         q.setIncludeScore(true);
         q.setRows(maxResults == 0 ? Integer.MAX_VALUE : maxResults);
-
+        if (returnFields != null) {
+          q.setFields(returnFields.length() > 0 ? returnFields : "*");
```

Comment:
https://lucene.apache.org/solr/6_6_0/solr-solrj/org/apache/solr/client/solrj/SolrQuery.html#setFields-java.lang.String...-
expects an Array of String instead of a CSV List !!!

Reply:
See https://lucene.apache.org/solr/6_6_0/solr-solrj/org/apache/solr/client/solrj/SolrQuery.html#setFields-java.lang.String...- ; it could be one or more Strings (and it works).